Java/src/main/java/net/hypixel/api/reply/AbstractReply.java

```diff
@@ -6,6 +6,9 @@
     protected boolean success;
     protected String cause;
 
+    transient protected int limitLeft;
+    transient protected int limitReset;
```

Comment:
Why are they transient. I understand that they're not importantly serializable properties, but it's just unnecessary. They should be final properties, even. It's not like they will/should change after the request has been made

Reply:
They are transient because this class gets deserialized with Gson, they would get ignored. I don't know if that matters tho if I set their values after. The reason they are not final is because I need my setters below to set these values as I don't have a constructor and don't have these values in json, ready to be deserialized. I wasn't sure I should rewrite the class and api-get function just to make these fields final. I could also add them in the json and let them be deserialized automatically, but that wouldn't be good practice as the names of these fields could change at any moment. Any ideas?

Reply:
I disagree. Letting GSON do the work of deserializing these would be perfectly fine in my opinion. We depend on the JSON structure because we are wrapping it. We cannot change this, it's fundamental as an API wrapper. For example, a while ago I made an issue over a property that was returning `null` every single time. It was just due to the JSON changing. It's part of the process, embrace it

Reply:
I understand what you're saying, I agree with this, my concern about the bad practice would be that you'd manually have to name the fields in the json object, thus creating two places that are connected and will need to be updated separately at the same time. This is feasible to work with but not the most elegant solution unless we wanna do something with reflection. Any ideas on that part?

Reply:
If I'm understanding correctly, you're saying you'd have to name them in the property names when extracting them such as `"ratelimit-remaining"` and also the field names to match (e.g. `rateLimitRemaining`). This can be solved by using GSON's `@SerializedName` annotation. Consider putting the names of these properties in a reasonable place as PSF Strings. Then simply:
```java
@SerializedName(REQUESTS_LEFT_FIELD)
private final int requestsLeft;

@SerializedName(LIMIT_RESET_FIELD)
private final int limitReset;
```

Reply:
I got this idea as well now, thank you for putting in this much effort, I will change this. Great input!

Reply:
No problem. Resolve the conversations